src/main/java/org/apache/ibatis/scripting/xmltags/ForEachSqlNode.java

```diff
@@ -154,7 +154,7 @@ public void appendSql(String sql) {
       GenericTokenParser parser = new GenericTokenParser("#{", "}", new TokenHandler() {
         @Override
         public String handleToken(String content) {
-          String newContent = content.replaceFirst("^\\s*" + item + "(?![^.,:\\s])", itemizeItem(item, index));
+          String newContent = content.replaceFirst("^\\s*" + item + "(?![^.\\[,:\\s])", itemizeItem(item, index));
           if (itemIndex != null && newContent.equals(content)) {
             newContent = content.replaceFirst("^\\s*" + itemIndex + "(?![^.,:\\s])", itemizeItem(itemIndex, index));
```

Comment:
@harawata Thanks for your review!
The `itemIndex` is variable name for representing a integer value. In other words, there is no case that it have a array object. Therefore I think it not need apply same change.

WDYT ?